src/v2i-hub/FLIRCameraDriverPlugin/src/FLIRPedestrianPresenceTrackingProcessor.cpp

```diff
@@ -125,10 +130,13 @@ namespace FLIRCameraDriverPlugin
             offsetY * std::sin(convertedCameraRotation * M_PI / 180.0);
 
         // Parse speed
-        if (!pr.get_child("speed").data().empty())
+        if (pr.get_optional<double>("speed"))
         {
             // Speed is in m/s
-            speed = std::stod(pr.get_child("speed").data());
+            speed = pr.get_optional<double>("speed").get();
+            if (speed > 0.0  && !heading_present) {
```

Comment:
I wonder if there will be a speed value smaller than epsilon that is still valid, like 0.00000001. Maybe it might be prudent to compare something like 0.001 or 0.01?